templates/common/alibabacloud/files/usr-local-bin-alibaba-kubelet-nodename.yaml

```diff
@@ -20,7 +20,6 @@ contents:
     INSTANCEID=$(curl -s ${META_EP}/instance-id)
 
    
-    cat > "${NODECONF}" <<EOF
-    [Service]
-    Environment="KUBELET_NODE_NAME=${REGIONID}.${INSTANCEID}"
+    cat > "${ENV_FILE}" <<EOF
+    KUBELET_NODE_NAME=${REGIONID}.${INSTANCEID}
```

Comment:
Related to the above, is it possible that `REGIONID` is unset because something else went wrong?

Reply:
Your suggested "strict mode" and "generator logging" changes uncovered the problem.

Here's the timeline:
1st generator run outputs this:
```
Sep 29 22:28:28 twiest-zxgz9-rhel-worker-us-east-1b-cbd33 unknown: BEGIN generator alibaba-kubelet-extra-env-generator.sh
Sep 29 22:28:28 twiest-zxgz9-rhel-worker-us-east-1b-cbd33 /usr/local/lib/systemd/system-generators/alibaba-kubelet-extra-env-generator.sh: line 26: cannot create temp file for here-document: Read-only file system
Sep 29 22:28:28 twiest-zxgz9-rhel-worker-us-east-1b-cbd33 systemd[733]: /usr/local/lib/systemd/system-generators/alibaba-kubelet-extra-env-generator.sh failed with exit status 1.
```

The generator eventually runs again, but it runs AFTER the kubelet service has started:

Kubetlet service start:
```
Sep 29 22:28:33 twiest-zxgz9-rhel-worker-us-east-1b-cbd33 systemd[1]: Starting Kubernetes Kubelet...
```

2nd generator run:
```
Sep 29 22:31:00 twiest-zxgz9-rhel-worker-us-east-1b-cbd33 unknown: BEGIN generator alibaba-kubelet-extra-env-generator.sh
Sep 29 22:31:00 twiest-zxgz9-rhel-worker-us-east-1b-cbd33 unknown: END generator alibaba-kubelet-extra-env-generator.sh
```


I'm going to work on a version of the generator that doesn't use a here-doc and re-run the test.

Thanks!